src/lib.rs

```diff
@@ -848,10 +860,117 @@ impl LineSpans for alloc::string::String {
     }
 }
 
+#[derive(Debug, Clone)]
+/// Owning pointer to a string that provides O(1) access to line slices
+///
+/// ```
+/// use line_span::CachedLines;
+/// let cache = CachedLines::without_ending(String::from("hello\nworld"));
+/// assert_eq!(&cache[0], "hello");
+/// assert_eq!(&cache[1], "world");
+/// ```
+///
+/// Slices might or might not include line breaks depending on a function used to construct it
+///
+/// - [`without_ending`](CachedLines::without_ending)
+/// - [`with_ending`](CachedLines::with_ending)
+///
+///
+#[cfg(feature = "alloc")]
+pub struct CachedLines {
+    content: alloc::string::String,
```

Comment:
Maybe this can use [`alloc::borrow::Cow`](https://doc.rust-lang.org/alloc/borrow/enum.Cow.html) instead to allow for borrowing a `&str` instead of needing to turn it into an owned `String`

(Not sure if it creates other issues)

```suggestion
pub struct CachedLines<'a> {
    content: alloc::borrow::Cow<'a>,
```

Reply:
I'm okay either way, but `Cow` adds complexity without adding significant benefits - `CachedLines` will have to allocate a vector of sizes regardless, adding a single allocation seems to be fine compared to the alternative of allocating and storing a whole bunch of lines.